src/vs/workbench/browser/actions/resizePart.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { TPromise } from 'vs/base/common/winjs.base';
+import nls = require('vs/nls');
+import { Registry } from 'vs/platform/platform';
+import { Action } from 'vs/base/common/actions';
+import { SyncActionDescriptor } from 'vs/platform/actions/common/actions';
+import { IWorkbenchActionRegistry, Extensions } from 'vs/workbench/common/actionRegistry';
+import { IPartService, Parts } from 'vs/workbench/services/part/common/partService';
+import { KeyMod, KeyCode } from 'vs/base/common/keyCodes';
+
+const RESIZE_INCREMENT = 50;
+
+export class ExpandViewAction extends Action {
+
+	public static ID = 'workbench.action.expandView';
+	public static LABEL = nls.localize('expandView', "Expand Current View");
+
+	constructor(
+		id: string,
+		label: string,
+		@IPartService private partService: IPartService,
+
+	) {
+		super(id, label);
+
+		this.enabled = !!this.partService;
+	}
+
+	public run(): TPromise<any> {
+
+		const isEditorFocus = this.partService.hasFocus(Parts.EDITOR_PART);
+		const isSidebarFocus = this.partService.hasFocus(Parts.SIDEBAR_PART);
+		const isPanelFocus = this.partService.hasFocus(Parts.PANEL_PART);
+
+		if (isSidebarFocus) {
+			return this.partService.resizePart(Parts.SIDEBAR_PART, RESIZE_INCREMENT);
+		}
+		else if (isPanelFocus) {
+			return this.partService.resizePart(Parts.PANEL_PART, RESIZE_INCREMENT);
+		}
+		else if (isEditorFocus) {
+			// console.log('editor f1ocus');
+			return this.partService.resizePart(Parts.EDITOR_PART, RESIZE_INCREMENT);
+		}
+		return TPromise.as(false);
+	}
+}
+
+export class ContractViewAction extends Action {
```

Comment:
@misoguy 
Agree, I've looked at your code a lot as well as comments. I was anxious
to get some initial feedback before doing more re- factoring.
I also think moving the resize actions into actions.ts like your code
is the newer approach. When I started I modeled my code after the toggleSidebarVisibility.ts
not sure why these several toggle functions are in separate files, anybody?

Hope you give it a try, seems like we have some common interests...